### PR TITLE
Disable debug code to avoid build issues

### PR DESCRIPTION
**Pull request by fabiencastan**

### CODE REVIEW

**Comment by simogasp** [comment]
Can u paste the building error for future memory?

**Comment by griwodz** [comment]
I fixed the code in the NDEBUG block instead of disabling it, but I agree with Simone. We should use define -DNDEBUG unless CMake is in Debug mode.

File: CMakeLists.txt [hunk 1]
```diff
@@ -117,7 +117,7 @@ if(CCTAG_WITH_CUDA)
     set(CUDA_NVCC_FLAGS_DEBUG   "${CUDA_NVCC_FLAGS_DEBUG};-G;-g")
   else()
     message(STATUS "Building in release mode")
-    add_definitions(-DNDEBUG)
+    add_compile_definitions(NDEBUG)
```
Comment (simogasp) [inline CMakeLists.txt]: CMake automatically add it to the flags if the build is release, no need to specify it

You can check here, this is the branch before you added it explicitly
https://travis-ci.org/alicevision/CCTag/jobs/563273927#L4118
and this is now
https://travis-ci.org/alicevision/CCTag/jobs/563423199#L4118
(it is set twice)

Comment (griwodz) [reply CMakeLists.txt]: Hmm. It appears to be added correctly when CMAKE_BUILD_TYPE is explicitly set to "Release" (which Travis is doing). If the CMAKE_BUILD_TYPE is left undefined, neither "Release" nor "Debug" flags are added.
I suppose that the better solution would be force a default CMAKE_BUILD_TYPE of either "Release" or "Debug"?

**Comment by griwodz** [comment]
Replaced the explicit -DNDEBUG with a default definition for CMAKE_BUILD_TYPE.

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,6 +18,15 @@ option(CCTAG_USE_POSITION_INDEPENDENT_CODE "Generate position independent code."
 option(CCTAG_ENABLE_SIMD_AVX2 "Enable AVX2 optimizations" OFF)
 option(BUILD_TESTS "Build the unity tests" ON)
 
+# Set a default build type if none was specified
```
Comment (fabiencastan) [inline CMakeLists.txt]: You can do it the same way that what is done in AliceVision:
https://github.com/alicevision/AliceVision/blob/develop/src/CMakeLists.txt#L78

Comment (fabiencastan) [reply CMakeLists.txt]: On Windows, as it generates the MSVC project which contains both release and debug, the CMAKE_BUILD_TYPE is not defined.

Comment (griwodz) [reply CMakeLists.txt]: I've done that and pushed it.

Comment (simogasp) [reply CMakeLists.txt]: What about something like that
'''
# Set a default build type if none was specified
set(default_build_type "Release")

if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
    message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE  STRING "Choose the type of build." FORCE)
    # Set the possible values of build type for cmake-gui
    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS  "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
endif()
'''

Plus the mvc stuff

Comment (fabiencastan) [reply CMakeLists.txt]: It's more code and more difficult to read, IMHO. Is there a difference in term of behavior?

I was suggesting to use the code from AV, as we are using it daily on win and linux without issue.
If we can do it better, we should update AV, CCTag, PopSift.

Comment (griwodz) [reply CMakeLists.txt]: I like to be able to switch between Release and Debug modes without removing the CMake cache, so for me, the 3 simple lines from AliceVision are the better option.

**Pull request merged**